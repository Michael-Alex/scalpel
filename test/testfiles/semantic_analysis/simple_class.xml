<namespace id="n0">
 <class name="square" id="c0">
  <constructor access="public">
   <parameters>
    <parameter name="size">
     <type>
      <cv_qualifiers const="true">
       <fundamental_type type="unsigned int"/>
      </cv_qualifiers>
     </type>
    </parameter>
   </parameters>
  </constructor>
  <destructor access="public" defined="false">
  </destructor>
  <simple_function name="size" access="public" const="true">
   <return_type>
    <fundamental_type type="unsigned int"/>
   </return_type>
  </simple_function>
  <simple_function name="size" access="public">
   <return_type>
    <fundamental_type type="void"/>
   </return_type>
   <parameters>
    <parameter name="size">
     <type>
      <cv_qualifiers const="true">
       <fundamental_type type="unsigned int"/>
      </cv_qualifiers>
     </type>
    </parameter>
   </parameters>
  </simple_function>
  <variable name="size_" access="private">
   <type>
    <fundamental_type type="unsigned int"/>
   </type>
  </variable>
 </class>
</namespace>
