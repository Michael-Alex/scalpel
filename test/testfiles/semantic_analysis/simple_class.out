{
  "id": 0,
  "classes":
  [
    {
      "id": 0,
      "name": "square",
      "constructors":
      [
        {
          "access": "public",
          "defined": true,
          "parameters":
          [
            {
              "name": "size",
              "type":
              {
                "cv-qualified type":
                {
                  "const": true,
                  "type":
                  {
                    "fundamental type": "unsigned int"
                  }
                }
              }
            }
          ]
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "simple functions":
      [
        {
          "id": 0,
          "name": "size",
          "access": "public",
          "const": true,
          "return type":
          {
            "fundamental type": "unsigned int"
          }
        },
        {
          "id": 1,
          "name": "size",
          "access": "public",
          "return type":
          {
            "fundamental type": "void"
          },
          "parameters":
          [
            {
              "name": "size",
              "type":
              {
                "cv-qualified type":
                {
                  "const": true,
                  "type":
                  {
                    "fundamental type": "unsigned int"
                  }
                }
              }
            }
          ]
        }
      ],
      "variables":
      [
        {
          "id": 0,
          "name": "size_",
          "access": "private",
          "type":
          {
            "fundamental type": "unsigned int"
          }
        }
      ]
    }
  ]
}
